mne/connectivity/cfc.py

```diff
@@ -0,0 +1,529 @@
+
+    # Compute filtered data
+    Ws = morlet(sfreq, freqs, n_cycles=n_cycles, sigma=None,
+                zero_mean=True)
+
+    parallel, my_abs_cwt, n_jobs = parallel_func(_abs_cwt, n_jobs)
+
+    if n_jobs == 1:
+        ampmat = np.empty((n_freqs, n_samples), dtype=np.float)
+        for e, W in enumerate(Ws):
+            ampmat[e] = np.array(_abs_cwt(data, W))
+    else:
+        ampmat = np.array(parallel(my_abs_cwt(data, W) for W in Ws))
+
+    # zscore for normalization
+    ampmat -= np.mean(ampmat, axis=1)[:, None]
+    ampmat /= np.std(ampmat, axis=1)[:, None]
+
+    # phase-triggered average of raw signal:
+    # and phase-triggered time-frequency amplitude values (normalized):
+    avg = np.zeros(times.size)
+    traces = np.zeros((n_freqs, times.size), dtype=np.float)
+    for ind in trigger_inds:
+        avg += data[0, ind - n_times_min: ind + n_times_max]
+        traces += ampmat[:, ind - n_times_min: ind + n_times_max]
+    avg /= len(trigger_inds)
+    traces /= len(trigger_inds)
+
+    return times, ampmat, traces, avg, trigger_inds
+
+
+def compute_cfc_stats(sfreq, ampmat, traces, freqs, trigger_inds,
+                      n_surrogates=10000, alpha=0.001, random_state=None):
+    """ Stats function for cross frequency coupling.
+
+    Parameters
+    ----------
+    sfreq : float
+        Sampling frequency.
+    ampmat : ndarray, shape (nfreq, n_samples)
+        Amplitude traces.
+    traces : ndarray, shape (n_epochs, n_times)
+        Normalized amplitude traces.
+    freqs : array, shape (n_freqs,)
+        Frequency points across range at which amplitudes are computed.
+    trigger_inds : array (trigger_inds,)
+        Indices of maxima in the phase signal.
+    n_surrogates : int
+        Number of surrogates. Default 10 ** 4.
+    alpha : float
+        Error rate allowed. Default 0.001.
+    random_state : None | int | instance of np.random.RandomState
+        np.random.RandomState to initialize the noise estimation.
+        Default value is None.
+
+    Returns
+    -------
+    ztraces : ndarray, shape (n_freqs, n_times)
+        Z-scored amplitude traces.
+    z_threshold : float
+        FDR corrected threshold for z-scored amplitude traces.
+    """
+
+    n_freqs = len(freqs)
+    n_samples = ampmat.size / n_freqs
+    epoch_length = traces.size / n_freqs
+
+    # permutation resampling for statistical significance:
+    # only need to compute mean and variance of surrogate distributions at one
+    # time point for all frequencies, since the surrogate trigger events could
+    # have occurred at any point
+    rng = check_random_state(random_state)
+    shifts = np.floor(rng.rand(2 * n_surrogates) * n_samples)
+    # get rid of trigger shifts within one epoch of actual occurence
+    shifts = shifts[shifts > epoch_length]
+    shifts = shifts[shifts < n_samples - epoch_length]
+
+    sur_distributions = np.zeros((n_freqs, n_surrogates), dtype=np.float)
```

Comment:
Rather than using a logger why not just include a progress bar here. (full disclosure, I love progress bars :))